### DISCUSSION TITLE
Feature: Using auth-kit.js without any react dependencies

### DISCUSSION THREAD

**Discussion by prince272** [react-auth-kit/react-auth-kit#1049] [Ideas]
**Is your feature request related to a problem? Please describe.**
Have you considered adding auth-kit.js without any react dependencies so that it could be used on most platform.
I thing the three issues that may ender the approach of developing the auth-kit are:

1. State management
2. window.localstorage found in TokenObject
3. cookies found in TokenObject.js

**Describe the solution you'd like**
With auth-kit, there should be no need for statement. Replace it with event that passes the current state of the token object.
eg: new AuthProvider ({authType: 'cookie' authName: '_auth', cookieDomain: window.location.hostname, cookieSecure: window.location.protocol === "https:")

**Describe alternatives you've considered**

**Additional context**
If you'd like me to work on the auth-kit of the react-auth-kit version. I'd be very happy. Also, the little problem is that I only know JavaScript and not typescript. But we'd work things out.

**Comment by prince272** [comment]
Also the strategy you used for refreshing tokens is great. It didn't internally rely on axios or any data fetching api.

**Comment by darkmatter18** [comment]
Hey @prince272 , Thanks for appreciating the project this much, and sorry for being this late. 
Your idea is awesome, it'll help a large no of communities.

But the problem with the current implementation is that it relies on React for state management. I don't have much clear idea of how to maintain the state without using React If you have any idea, let me know.

Let's start the discussion and let's build this together.

**Comment by prince272** [comment]
First of all, let me express my gratitude for your reply.

**My Research after posting this issue:** I did some research and discovered several approaches for managing the tokens using axois  and also discovered a way to support both client and server-side, After that, I wrote my own custom solution at https://github.com/prince272/CleanArchitecture/blob/master/src/CleanArchitecture.Client/utils/axios.js which uses BehaviorSubject for rasing the user changed event and for containing the user object as well, and cookies for storing the tokens and user object too. 

Usage: https://github.com/prince272/CleanArchitecture/blob/master/src/CleanArchitecture.Client/client.js

**Your solution:** You can maintain the state using a dependency library:  `import { BehaviorSubject } from 'rxjs';` You can learn more about BehaviorSubject at https://www.learnrxjs.io/learn-rxjs/subjects/behaviorsubject
If you can write your own behaviorsubject so that your library does not depend on any external resource, thats fine.

Also, you can review my current solution at https://github.com/prince272/CleanArchitecture/tree/master/src/CleanArchitecture.Client
